packages/mongo-nav/src/mongo-select/OrgSelect.tsx

```diff
@@ -208,17 +233,49 @@ function OrgSelect({
       return search.test(datum.orgName);
     });
 
-    return filtered;
+    setFilteredData(filtered);
   };
 
-  const renderedData = onChangeProp ? consumerFilteredData : filterData();
+  const didMount = useRef(false);
+  useEffect(() => {
+    // defer all behavior to user-provided filtering
+    if (onChangeProp) {
+      return;
+    }
+
+    // clear filtering if the search term is cleared
+    if (value === '') {
+      setFilteredData(data);
+      return;
+    }
+
+    // serverside filtering (global read only)
+    if (admin) {
```

Comment:
This is breaking in Dev mode. Might want to add `if (admin && mode !== 'dev')` or something like that

Reply:
Ah, this wasn't something that I assumed we wanted to handle, but is an easy fix!